library/cc/headers_builder.cc

```diff
@@ -3,20 +3,20 @@
 namespace Envoy {
 namespace Platform {
 
-HeadersBuilder& HeadersBuilder::add(const std::string& name, const std::string& value) {
+HeadersBuilder& HeadersBuilder::add(std::string name, std::string value) {
   if (this->isRestrictedHeader(name)) {
     return *this;
   }
-  this->headers_[name].push_back(value);
+  this->headers_[std::move(name)].push_back(std::move(value));
```

Comment:
Moving the key (here and elsewhere) is kinda interesting. If they key doesn't exist in the map then it's optimal. If it does then we've created an unnecessary copy and string_view would have been better.

I don't have any particular opinions here, so I'm not opposed to taking the key by value and moving it like this. Intuitively it seems like users would most likely be calling these methods new keys that aren't already in the map, in which case this new change is ideal. But I don't have any evidence for my intuition so I'll leave it up to you/others.

Reply:
Yeah, I debated this myself. My intuition matches yours that this is likely called when the key is not in the map, but I don't feel strongly. Let's see what the other Envoy Mobile reviewers have to say?